File: code/SWE2-Project/Animal Tracker/src/test/java/edu/westga/cs3211/animaltracker/model/ProjectModel/TestAddAnimal.java
package edu.westga.cs3211.animaltracker.model.ProjectModel;

import edu.westga.cs3211.animaltracker.model.*;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.LinkedList;

import static org.junit.jupiter.api.Assertions.*;

class TestAddAnimal {
    private Project project;
    @BeforeEach
    void setUp() {
        DataStorage.reset();
        this.project = new Project(new ArrayList<>(), "Test Project", new ArrayList<>());
    }

    @Test
    void testAddNullAnimal() {
        assertThrows(IllegalArgumentException.class, () -> {
            this.project.addAnimal(null);
        });
    }

    @Test
    void testAddValidAnimal() {
        var animal = new Animal(AnimalClass.BIRD, 11.0, 15.0, 17.0, 122345, "");
        this.project.addAnimal(animal);

        assertAll(() -> {
            assertTrue(this.project.containsAnimal(animal));
            assertEquals(1, this.project.getAnimals().size());
            assertEquals(animal, this.project.getAnimals().getFirst());
            assertTrue(this.project.containsAnimalType(AnimalClass.BIRD));
        });
    }

    @Test
    void testAddExisingAnimal() {
        var animals = new LinkedList<Animal>();
        var animal = new Animal(AnimalClass.BIRD, 11.0, 15.0, 17.0, 122345, "");
        animals.add(animal);
        for (var currAnimal : animals) {
            this.project.addAnimal(currAnimal);
        }

        assertThrows(IllegalArgumentException.class, () -> {
            project.addAnimal(animal);
        });
    }
}

File: code/SWE2-Project/Animal Tracker Server/src/test/model_test/project_test/test_add_animal.py
import unittest

from model.Animal import Animal
from model.AnimalClass import AnimalClass
from model.Project import Project


class TestAddAnimal(unittest.TestCase):
    def test_add_none_animal(self):
        project = Project("1234", set(), set(), 1)
        with self.assertRaises(Exception):
            project.add_animal(None)

    def test_add_duplicate_animal(self):
        project = Project("1234", set(), set(), 1)
        animal = Animal(AnimalClass.BIRD, 20, 20, 20, 20, "1234")
        project.add_animal(animal)
        with self.assertRaises(Exception):
            project.add_animal(animal)

    def test_add_one_animal(self):
        project = Project("1234", set(), set(), 1)
        animal = Animal(AnimalClass.BIRD, 20, 20, 20, 20, "1234")
        project.add_animal(animal)
        self.assertEqual(len(project.get_animals()), 1)
        self.assertTrue(animal in project.get_animals())

    def test_multiple_animals(self):
        project = Project("1234", set(), set(), 1)
        animal = Animal(AnimalClass.BIRD, 20, 20, 20, 20, "1234")
        second_animal = Animal(AnimalClass.BIRD, 223, 15, 260, 2123, "1234")
        project.add_animal(animal)
        project.add_animal(second_animal)
        self.assertEqual(len(project.get_animals()), 2)
        self.assertTrue(animal in project.get_animals())
        self.assertTrue(second_animal in project.get_animals())

